src/main/java/desi/juan/email/api/Email.java

```diff
@@ -23,104 +25,245 @@
  */
 package desi.juan.email.api;
 
-import com.google.common.collect.ImmutableMultimap;
-import com.google.common.collect.Multimap;
+import com.google.common.collect.ImmutableList;
+import desi.juan.email.internal.OutgoingEmail;
 
+import javax.mail.Folder;
+import javax.mail.Header;
 import java.time.LocalDateTime;
-import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 
+import static desi.juan.email.api.EmailConstants.NO_SUBJECT;
+import static java.lang.String.format;
+
 /**
- * Generic contract for email messages implementations.
+ * Generic contract for Email implementations.
  */
-public interface Email {
+public abstract class Email {
```

Comment:
we should maintain the interface, you can create an AbstractEmail that still follows the interface